src/batBitSet.ml

```diff
@@ -48,7 +48,7 @@ let capacity t = (Bytes.length !t) * 8
 let empty () = ref (Bytes.create 0)
 
 let create_ sfun c n = (* n is in bits *)
-  if n < 0 then invalid_arg ("BitSet."^sfun^": negative size");
+  if n < 0 then invalid_arg ("BitSet." ^ sfun ^ ": negative size");
```

Comment:
While we are at it, I think that `Printf.ksprintf invalid_arg "Bitset.%s: negative size" sfun` would be nicer (and possibly more efficient) here.

Reply:
How about including an actually received `n`? Over years, I've developed a practice that involves stuffing as much information into Invalid_argument messages, because it's an investment that pays off easily. Instead of "Invalid index", my functions throw "Invalid index (expected n >= 0 && n < %d, but got %d)". As for performance, no sane code should ever catch Invalid_argument for normal control flow, as it's akin to a dynamic assertion. I even think of opening an issue about reworking whole Batteries in this style.

Reply:
> As for performance, no sane code should ever use Invalid_argument for normal control flow, as it's akin to a dynamic assertion.

Typo: "as it's akin to _suppressing_ a dynamic assertion". I talk about a scenario when the user prefers to catch Invalid_argument instead of ensuring that no bad argument ever gets into the function. My position is that Invalid_argument is about as bad as Assert_failure -- one should never catch it explicitly.

Reply:
I agree with indicating the out-of-bound index in the error message. I thought about it during my review and decided not to add more work, but if you are voluntary...